translate-c: handle nested statement expressions

ehaas opened the pull request:
When translating statement expressions, look up the block label
in the scope instead of hardcoding "blk", since they may be
nested.

Vexu commented:
I already fixed this in #7479, could you make this pr add a test case to it?

ehaas commented:
Not sure I fully understand what you're asking - would you like me to open a PR against `translate-c-ast` with this test case? or something else?

Vexu commented:
No need for a new pr, you can just rebase this to `translate-c-ast` and then change this pr's base branch.

ehaas commented:
Ok cool...if you don't mind waiting until tomorrow, I can do it then. It's super late where I am .

ehaas commented:
@Vexu i updated this branch as requested.

Are there any special instructions for getting `translate-c-ast` to compile? I'm using "Option A" from here https://github.com/ziglang/zig/wiki/Building-Zig-From-Source on MacOS and I get errors like this (doing a completely clean build, deleted zig-cache etc)

```
[ 99%] Building self-hosted component /Users/ehaas/source/zig/build/zig1.o
./src/Module.zig:1244:100: error: missing field: 'cc'
    const fn_type_inst = try astgen.addZIRInst(mod, &fn_type_scope.base, fn_src, zir.Inst.FnType, .{
                                                                                                   ^
./src/Module.zig:1244:47: note: referenced here
    const fn_type_inst = try astgen.addZIRInst(mod, &fn_type_scope.base, fn_src, zir.Inst.FnType, .{
                                              ^
./src/Module.zig:951:54: note: referenced here
    const type_changed = mod.astgenAndSemaDecl(decl) catch |err| switch (err) {
                                                     ^
./src/Module.zig:1499:43: error: no member named 'mut_token' in struct 'std.zig.ast.VarDecl'
    const is_mutable = token_tags[var_decl.mut_token] == .keyword_var;
                                          ^
./src/astgen.zig:677:26: error: container 'std.zig.ast.Node' has no member called 'labeled_block'
    block_node: *ast.Node.labeled_block,
                         ^
./src/astgen.zig:471:16: note: referenced here
        return labeledBlockExpr(mod, parent_scope, rl, node, .block_comptime);
               ^
```

Vexu commented:
`astgen` is not done yet so you need to build without stage2 by passing `-DZIG_OMIT_STAGE2=on` to cmake.

ehaas commented:
That worked, thanks! I have a couple other local branches with translate-c improvements - should I port them to PRs against this branch? Or just wait until it's merged into master?

Vexu commented:
#7920 will likely be merged quite soon so I'd probably just wait for that.
